cmd/krew/cmd/namingutils.go

```diff
@@ -48,3 +54,23 @@ func canonicalName(p index.Plugin, indexName string) string {
 	}
 	return indexName + "/" + p.Name
 }
+
+// allIndexes returns a slice of index name and URL pairs
+func allIndexes(p environment.Paths) ([]indexoperations.Index, error) {
+	indexes := []indexoperations.Index{
+		{
+			Name: constants.DefaultIndexName,
+			URL:  constants.DefaultIndexURI,
+		},
+	}
+	if os.Getenv(constants.EnableMultiIndexSwitch) != "" {
+		out, err := indexoperations.ListIndexes(p)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to list plugin indexes available")
+		}
+		if len(out) != 0 {
+			indexes = out
+		}
+	}
+	return indexes, nil
+}
```

Comment:
It might've been better if we extracted the feature gate into `indexoperations.ListIndexes` so we don't reimplement it here.

IIRC there's at least one more place we hardcode this:

```go
[]indexoperations.Index{
		{
			Name: constants.DefaultIndexName,
			URL:  constants.DefaultIndexURI,
		},
```

you should be able to find very easily.

Reply:
yeah if you handle it in one place (for now, until we remove the feature gate saves the following +60 lines in `namingutils_test.go` for us, and reduces duplication)

Reply:
Ah good call, I didn't think about that.

Reply:
So this was something that we talked about when going over the proposal:
> If no indexes are configured, a message is printed explaining how to add krew-index as default

I just want to make sure we're cool with that. The current functionality is that krew-index gets added by default, but once multi-index is out that means that users won't be able to `krew install ctx` right after install.

Reply:
> You should hardcode that index to that method only when X_KREW_MULTI_INDEX is "not" set.

We decided before that `krew index list` after a fresh install should work and return an empty list. Do you want me to change that behavior and accompanying tests if we want it to return the hardcoded index when env var isn't set?

> How do we add the default=krew-index (once multi-index work is complete) is another discussion which I'm tracking at #566. (Right now we're auto-adding it if it's missing on install/update/upgrade) but this is a side-effect of old code, and it's not about this PR.

I brought it up because that was part of the proposal and it seemed like this would be the PR to take care of the help message when no indexes are configured. If that's not the case then what should the behavior be when the env var is set, but no indexes are configured?

Reply:
> We decided before that krew index list after a fresh install should work and return an empty list. Do you want me to change that behavior and accompanying tests if we want it to return the hardcoded index when env var isn't set?

We should keep this behavior. I only wanted that we call indexoperations.ListIndexes always, but without the feature flag, it returns default index, with the feature flag it lists from filesystem.

I can make this change if you want me to.

> I brought it up because that was part of the proposal and it seemed like this would be the PR to take care of the help message when no indexes are configured

That can be a separate PR, let's keep this one small.

Reply:
Ok I can take care of it here 👍